### DISCUSSION TITLE
LiquidSoap log ; Mad.End_of_stream - What is it?

### DISCUSSION THREAD

**Discussion by SpeedyCharly** [AzuraCast/AzuraCast#4817] [Q&A] [answered]
I often encounter this line in my Liquidsoap log (highlighted in red);
![Image2](https://user-images.githubusercontent.com/66107794/143531649-78bd9f99-521d-49ef-a840-36f7c76fd34d.jpg)

Does that mean that the file is somehow trunckated and ends abruptly before the previously declared duration? The next song is immediately queued following this, but I fear that it may negatively impact listener enjoyment.  What is actually going on here, and how can it be prevented if it is, in fact, an issue?

**Comment by Vaalyn** [comment, marked as answer]
This is a log entry from the Liquidsoap [decoder implementation](https://github.com/savonet/liquidsoap/blob/b69b0352cc7258c44ce4392386cd60c15f07f6a1/src/decoder/mad_decoder.ml) for the [`MAD: MPEG Audio Decoder`](https://www.underbit.com/products/mad) for mp3's that logs that the decoder has ended the decoding because it reached the end of the audio stream of the file.

**Reply by SpeedyCharly** [reply]
Thanks Vaalyn - so no issue then, all is good.
